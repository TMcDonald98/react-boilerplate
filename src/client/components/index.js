
export App from './App/App';
export Switcher from './Switcher/Switcher';

// Icons
export CheckIcon from './Icon/Check';
export XIcon from './Icon/X';
