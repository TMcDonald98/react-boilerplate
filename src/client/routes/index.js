
export * from './routesMap';
export Home from './Home/Home';
