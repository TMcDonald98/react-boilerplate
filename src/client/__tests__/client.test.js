import { Client as _Client } from '../client';
import axios from 'axios';

jest.mock('state', () => ({
  store: {
    getState: jest.fn()
  }
}));
jest.mock('axios');
jest.mock('state/selectors', () => ({
  getApiKey: () => 1
}));
axios.mockImplementation((...arg) => arg);

const Client = new _Client('test');

describe('client', () => {
  describe('class methods', () => {
    describe('request', () => {
      it('should fire a request', () => {
        expect(Client.request(Client.getPlaceholder())).toMatchSnapshot('request');
      });
    });

    describe('addAuthData', () => {
      it('should add auth data', () => {
        expect(Client.addAuthData(true)).toMatchSnapshot('addAuthData::true');
      });

      it('should not add auth data', () => {
        expect(Client.addAuthData(false)).toMatchSnapshot('addAuthData::false');
      });
    });

    describe('addApiKey', () => {
      it('should return the right object', () => {
        expect(Client.addApiKey()).toMatchSnapshot('addApiKey');
      });
    });
  });
});
