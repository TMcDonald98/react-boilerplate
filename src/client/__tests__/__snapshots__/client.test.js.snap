// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`client class methods addApiKey should return the right object: addApiKey 1`] = `
Object {
  "headers": Object {
    "API-KEY-HEADER-NAME": undefined,
  },
}
`;

exports[`client class methods addAuthData should add auth data: addAuthData::true 1`] = `
Object {
  "headers": Object {
    "API-KEY-HEADER-NAME": 1,
  },
}
`;

exports[`client class methods addAuthData should not add auth data: addAuthData::false 1`] = `Object {}`;

exports[`client class methods request should fire a request: request 1`] = `
Array [
  Object {
    "0": Object {
      "data": undefined,
      "method": "GET",
      "params": Object {
        "limit": 30,
      },
      "url": "undefined/v1/placeholder",
    },
    "data": undefined,
    "headers": Object {
      "API-KEY-HEADER-NAME": 1,
    },
    "method": undefined,
    "url": NaN,
  },
]
`;
